plenum/test/replica/test_instance_faulty_processor.py

```diff
@@ -0,0 +1,409 @@
+import pytest
+
+from plenum.common.messages.node_messages import BackupInstanceFaulty
+from plenum.common.types import f
+from plenum.server.backup_instance_faulty_processor import BackupInstanceFaultyProcessor
+from plenum.server.quorums import Quorums
+from plenum.server.replica import Replica
+from plenum.server.suspicion_codes import Suspicions
+from plenum.test.primary_selection.test_primary_selector import FakeNode
+
+
+class FakeReplicas:
+    def __init__(self, node, replicas):
+        self._replicas = replicas
+        self._node = node
+        self.remove_replica_calls = []
+        self.add_replica_calls = []
+
+        self.items = lambda: replicas.items()
+        self.keys = lambda: replicas.keys()
+
+    def remove_replica(self, inst_id):
+        self.remove_replica_calls.append(inst_id)
+        self._replicas.pop(inst_id)
+
+    def add_replica(self, inst_id):
+        self.add_replica_calls.append(inst_id)
+        self._replicas.update(inst_id=Replica(node=self._node, instId=inst_id))
+
+    def __getitem__(self, item):
+        return self._replicas[item]
+
+
+@pytest.fixture(scope="function")
+def backup_instance_faulty_processor(tdir, tconf):
+    node = FakeNode(tdir, config=tconf)
+    node.view_change_in_progress = False
+    node.requiredNumberOfInstances = len(node.replicas)
+    node.allNodeNames = ["Node{}".format(i)
```

Comment:
Should it be `"Node{}".format(i+1)`?

Reply:
We have a `range(1, (node.requiredNumberOfInstances - 1) * 3 + 2)` . But I will change it to `range((node.requiredNumberOfInstances - 1) * 3 + 1)`  to make it code more understandable.